### ISSUE TITLE
PowerPoint API getSelectedDataAsync with coersion type SlideRange fails on new line character in slide title.

### ISSUE THREAD

**Issue by CaptainTammy** [OfficeDev/office-js#4200]
When using getSelectedDataAsync with coersiontype SlideRange the request fails if there is a new line (\n) character in the title of the slide. With failing I mean it is not giving an error or failed status in the callback or console log, there is simply nothing returned and the callback function is never reached. This makes the rest of my code in the callback to not run.

Changing the \n to \r in the slide title text seems to be working, but this is not an option as I can't know what the user enters in the slide title.

There seems to be an error in the API code when escaping new line characters for the slide title when calling getSelectedDataAsync.

I would expect all new line characters to work in all scenarios, and the callback to be reached when doing the getSelectedDataAsync request.

Here's an example I've been testing with.
[Taskpane.zip](https://github.com/OfficeDev/office-js/files/14458229/Taskpane.zip)

**Comment by AkhileshShah-MS** [comment]
Hi @CaptainTammy  ! Thanks for reaching out to us.
Looping in @EsterBergen to help with the issue.
Thanks!

**Comment by EsterBergen** [comment]
@CaptainTammy - thanks for submitting. We'll look into this and respond soon.

**Comment by EsterBergen** [comment]
@CaptainTammy -Is this specific to one platform or a version?
Can you provide details about the environment?

**Your Environment**
Platform [PC desktop, Mac, iOS, Office on the web]:
Host [Excel, Word, PowerPoint, etc.]: Powerpoint
Office version number: 
Operating System: 
Browser (if using Office on the web):

**Comment by CaptainTammy** [comment]
@EsterBergen 
I don't think this is platform specific as we have encountered it on both Windows and Mac versions while developing. 

Environment
Platform [PC desktop, Mac, iOS, Office on the web]: PC, Mac
Host [Excel, Word, PowerPoint, etc.]: Powerpoint
Office version number: Mac 16.82, Windows 2401
Operating System: MacOS, Windows
Browser (if using Office on the web): NA

**Comment by andriy101** [comment]
Here are the repro steps:

- Create presentation with one slide, which have a tile text box as first shape.
- Add any web based add-in, for example web viewer
- In add-in's console run: 

```
PowerPoint.run(async context => {
    const slide = context.presentation.slides.getItemAt(0);
    slide.shapes.load('items');
    await context.sync();
    const shape = slide.shapes.items.at(0);
    shape.load('*');
    await context.sync();
    shape.textFrame.textRange.text = 'FIRST-LINE\nSECOND-LINE';
});
```
- Previous code should insert header **FIRST-LINE\nSECOND-LINE** in two lines
- Now try to read it, run in console:
```
Office.context.document.getSelectedDataAsync(Office.CoercionType.SlideRange, asyncResult => {
    if (asyncResult.status === Office.AsyncResultStatus.Failed) {
        console.log('ERR');
    } else {
        console.log('SUCCESS', asyncResult.value.slides[0]);
    }
});
```
- no console log will be printed, code is stuck and nothing happens
- if I use \r instead of \n, everything works as expected.

Some resource: https://www.rdpslides.com/pptfaq/FAQ00992_Paragraph_endings_and_line_breaks.htm

I wonder, if we may have similar issues in another Office APIs...
